starknet_py/utils/typed_data.py

```diff
@@ -341,6 +438,14 @@ def _prepare_merkle_tree_root(self, value: List, context: TypeContext) -> int:
         return MerkleTree(struct_hashes, self._hash_method).root_hash
 
     def _get_merkle_tree_leaves_type(self, context: TypeContext) -> str:
+        target_type = self._resolve_type(context)
+
+        if not isinstance(target_type, MerkleTreeParameter):
+            raise ValueError("Missing 'contains' field in target type.")
```

Comment:
Is it correct error message?

Reply:
I've changed it to `Target type is not a merkletree type.` 👍